core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -276,7 +275,8 @@ private static Collection<PartitionStats> computeAndMergeStatsIncremental(
     PartitionMap<PartitionStats> statsMap = PartitionMap.create(table.specs());
     // read previous stats, note that partition field will be read as GenericRecord
     try (CloseableIterable<PartitionStats> oldStats =
-        readPartitionStatsFile(schema(partitionType), Files.localInput(previousStatsFile.path()))) {
+        readPartitionStatsFile(
+            schema(partitionType), table.io().newInputFile(previousStatsFile.path()))) {
```

Comment:
Just using the FileIO as pointed out in the PR. 
Unrelated to this bug fix. But related to this feature.